### ISSUE TITLE
If no group is defined on SerializationContext all fields with exposed true for groups are exposed

### ISSUE THREAD

**Issue by mamartins** [schmittjoh/JMSSerializerBundle#491]
Hello,

I was trying to have 2 expose methods for my user entity, one for the owner (with hashkey etc.) and other for "public" usage with only it's non sensitive info.

I've done that with groups, the problem is that if no group is set on the controller it exposes all methods even those that have the expose true only for a specific group.

eg:

```

AppBundle\Entity\User:
    exclusion_policy: ALL
    properties:
        referralCode:
            expose: true
        referrer:
            expose: true
        apikey:
            expose: true
            groups: [owner]
```

I expected that when no `$context = SerializationContext::create()->setGroups(array("Default", "owner"));` was set I got only the properties referralCode and referrer, but the apikey also cames along.

Having to set SerializationContext on all methods to hide the apikey can lead to potential security flaw if you miss to add it in some place. When no group is defined it should only expose the properties that don't have that restriction

**Comment by fliespl** [comment]
@mamartins have you found solution to this issue?

**Comment by fliespl** [comment]
Nevermind. Found out that setting below does the magic :)

fos_rest:
    serializer:
        groups: [Default]

**Comment by akomm** [comment]
Well yes, `fos_rest` has a solution for this, but this is `jms_serializer`. I'v ditched `fos_rest` for reasons. 

[This](https://github.com/schmittjoh/serializer/blob/f7e2818b8fcc71826c5743dd721bc0cf5bb30868/src/JMS/Serializer/Serializer.php#L74-L78) is a problem. It does not allow us to provide custom default SerializationContext.

An simple solution (not for this bundle) would be this:

```yml
parameters:
  jms_serializer.serializer.class: 'AppBundle\Serializer\CustomContextSerializer'
```

```php
<?php

namespace AppBundle\Serializer;

use JMS\Serializer\SerializationContext;
use JMS\Serializer\Serializer;

/**
 * Class CustomContextSerializer
 * @package AppBundle\Serializer
 */
class CustomContextSerializer extends Serializer
{
    public function serialize($data, $format, SerializationContext $context = null)
    {
        if (null == $context) {
            $context = new SerializationContext();
            $context->setGroups(['Default']);
        }

        return parent::serialize($data, $format, $context);
    }
}
```

To make it configurable you would also need to extend the JMS Serializer and add a constructor configuration parameter, which is injected based on extension-configuration. Then override the serialize method and instantiate a context with the defaults provided via configuration.

**Comment by goetas** [comment]
With context factories https://github.com/schmittjoh/serializer/pull/645 added in jms serializer v1.4 and jms serializer bundle v1.2 , you can provide a factory for the context

**Comment by iisisrael** [comment]
When using this bundle, you now don't need to create your own context factory, as it can be done for you via configurations.  Override the default behavior to exclude a property from the Default context if one or more groups is set on the property by using this configuration (available in versions ^1.4.0 with [PR #556](https://github.com/schmittjoh/JMSSerializerBundle/pull/556)):
```
jms_serializer:
    default_context:
        serialization:
            groups: ['Default']
```
To then _include_ a property in the Default context as well as another group context (still excluding it from any other group context), add it explicitly to the property groups:
```
    exclusion_policy: ALL
    properties:
        someProperty:
            expose: true
            groups: [ 'mygroup', 'Default' ]
```

**Comment by lukepass** [comment]
@fliespl Thank you. Is this still the only solution?
@iisisrael I didn't catch your last comment. Is it a direct solution?
